test/system/src/test/java/io/pravega/test/system/MultiReaderWriterWithFailOverTest.java

```diff
@@ -0,0 +1,423 @@
+        eventReadCount = new AtomicLong(); // used by readers to maintain a count of events.
+
+        //get ClientFactory instance
+        log.info("Scope passed to client factory {}", scope);
+        clientFactory = new ClientFactoryImpl(scope, controller);
+        log.info("Client factory details {}", clientFactory.toString());
+        //create writers
+        log.info("Creating {} writers", NUM_WRITERS);
+        List<EventStreamWriter<Long>> writerList = new ArrayList<>();
+        log.info("Writers writing in the scope {}", scope);
+        for (int i = 0; i < NUM_WRITERS; i++) {
+            log.info("Starting writer{}", i);
+            final EventStreamWriter<Long> writer = clientFactory.createEventWriter(STREAM_NAME,
+                    new JavaSerializer<Long>(),
+                    EventWriterConfig.builder().build());
+            writerList.add(writer);
+        }
+
+        //create a reader group
+        log.info("Creating Reader group : {}", readerGroupName);
+        log.info("Scope passed to readergroup manager {}", scope);
+        readerGroupManager = ReaderGroupManager.withScope(scope, controllerUri);
+        readerGroupManager.createReaderGroup(readerGroupName, ReaderGroupConfig.builder().startingTime(0).build(),
+                Collections.singleton(STREAM_NAME));
+        log.info("Reader group name {} ", readerGroupManager.getReaderGroup(readerGroupName).getGroupName());
+        log.info("Reader group scope {}", readerGroupManager.getReaderGroup(readerGroupName).getScope());
+        log.info("Online readers {}", readerGroupManager.getReaderGroup(readerGroupName).getOnlineReaders());
+
+        //create readers
+        log.info("Creating {} readers", NUM_READERS);
+        List<EventStreamReader<Long>> readerList = new ArrayList<>();
+        String readerName = "reader" + new Random().nextInt(Integer.MAX_VALUE);
+        log.info("Scope that is seen by readers {}", scope);
+        //start reading events
+        for (int i = 0; i < NUM_READERS; i++) {
+            log.info("Starting reader{}", i);
+            log.info("Creating reader with id {}", readerName + i);
+            final EventStreamReader<Long> reader = clientFactory.createReader(readerName + i,
+                    readerGroupName,
+                    new JavaSerializer<Long>(),
+                    ReaderConfig.builder().build());
+            readerList.add(reader);
+        }
+
+        log.info("Test with 2 controller, SSS instances running and without a failover scenario");
+        // start writing asynchronously
+        CompletableFuture<Void> writerFuture = CompletableFuture.runAsync(() -> {
+            writerList.forEach(writer -> startWriting(eventData, writer));
+        });
+        //start reading asynchronously
+        CompletableFuture<Void> readerFuture = CompletableFuture.runAsync(() -> {
+            readerList.forEach(reader -> startReading(eventsReadFromPravega, eventData, eventReadCount, stopReadFlag, reader));
```

Comment:
Same as the writers, if we want parallel readers we may have to create List<CompletableFuture>